modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/ConfigParserTestCase.java

```diff
@@ -111,6 +112,19 @@ public void sameInstanceForEquivalentEvent() throws Exception
     }
 
     @Test
+    public void sameInstanceConfigDifferentExpressionValuesForDifferentEvents() throws Exception
+    {
+        MuleEvent event = getHeisenbergEvent();
```

Comment:
also, please add a test which verifies that a dynamic config is still dynamic even if it has a function